.github/workflows/tests.yml

```diff
@@ -111,3 +111,99 @@ jobs:
           EVAP_SKIP_CHECKOUT: true
       - name: Reload backup
         run: echo "yy" | deployment/load_production_backup.sh backup.json
+
+  compile_scss:
+    runs-on: ubuntu-18.04
+
+    name: Compile Scss
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Install dart-sass
+        run: deployment/install_dart_sass.sh
+      - name: Compile Scss
+        run: sass evap/static/scss/evap.scss evap/static/css/evap.css
+      - name: Store Css
+        uses: actions/upload-artifact@v2
+        with:
+          name: css
+          path: evap/static/css/evap.css
+
+
+  generate_fixtures:
+    runs-on: ubuntu-18.04
+
+    container:
+      image: python:3.7
+
+    name: Generate Typescript fixtures
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: evap
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+      redis:
+        image: redis
+        options: --health-cmd "redis-cli ping" --health-interval 10s --health-timeout 5s --health-retries 5
+        ports:
+          - 6379:6379
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: pip install -r requirements-test.txt
+      - name: Add localsettings
+        run: cp evap/settings_test.py evap/localsettings.py
+      - name: Generate fixtures
+        run: coverage run manage.py ts generate_fixtures && codecov -X gcov -F generate-fixtures
+      - name: Store fixtures
+        uses: actions/upload-artifact@v2
+        with:
+          name: fixtures
+          path: evap/static/ts/fixtures
+
+
+  typescript:
+    runs-on: ubuntu-18.04
+
+    needs: [ compile_scss, generate_fixtures ]
+
+    name: Test Typescript
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Install packages
+        run: sudo apt-get install xvfb libgbm1
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: "15.4.0"
+      - name: Install Node dependencies
+        run: npm ci
+      - name: Compile Typescript
+        run: npx tsc --project evap/static/ts/tsconfig.compile.json
```

Comment:
Just as with node, you could add a action to install python on this container: https://github.com/actions/setup-python without switching the container we are based on

Reply:
Hm, that would be indeed a possibility, but I currently see some unclearnesses:
- The new `management.py ts test` command bundles several other commands which should be separate steps in the CI (for display purposes). Okay, we could an option for it (like `--skip-setup`), but this also feels a bit off.
- The other actions use the container and this would be inconsistent ;)